nf-modules-template/.github/{% if ci =='github' %}workflows{% endif %}/nf-test.yml.jinja

```
@@ -0,0 +1,201 @@
+    paths-ignore:
+      - "**/meta.yml"
+  merge_group:
+    types: [checks_requested]
+    branches: [{{ default_branch }}]
+  workflow_dispatch:
+
+# Cancel if a newer run is started
+concurrency:
+  group: {% raw %}${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}{% endraw %}
+  cancel-in-progress: true
+
+env:
+  GITHUB_TOKEN: {% raw %}${{ secrets.GITHUB_TOKEN }}{% endraw %}
+  # renovate: datasource=github-releases depName=askimed/nf-test versioning=semver
+  NFT_VER: "0.9.2"
+  NXF_ANSI_LOG: false
+  NXF_SINGULARITY_CACHEDIR: {% raw %}${{ github.workspace }}/.singularity{% endraw %}
+  NXF_SINGULARITY_LIBRARYDIR: {% raw %}${{ github.workspace }}/.singularity{% endraw %}
+  # renovate: datasource=github-releases depName=nextflow/nextflow versioning=semver
+  NXF_VER: "25.04.0"
+
+jobs:
+  nf-test-changes:
+    name: nf-test-changes
+    runs-on: ubuntu-latest
+    outputs:
+      # Expose detected tags as 'modules' and 'workflows' output variables
+      paths: {% raw %}${{ steps.list.outputs.components }}{% endraw %}
+      modules: {% raw %}${{ steps.components.outputs.modules }}{% endraw %}
+      subworkflows: {% raw %}${{ steps.components.outputs.subworkflows}}{% endraw %}
+      shard: {% raw %}${{ steps.set-shards.outputs.shard }}{% endraw %}
+      total_shards: {% raw %}${{ steps.set-shards.outputs.total_shards }}{% endraw %}
+    steps:
+      - name: Clean Workspace # Purge the workspace in case it's running on a self-hosted runner
+        run: |
+          ls -la ./
+          rm -rf ./* || true
+          rm -rf ./.??* || true
+          ls -la ./
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5
+        with:
+          fetch-depth: 0
+
+      - name: List nf-test files
+        id: list
+        uses: adamrtalbot/detect-nf-test-changes@de3c3c8e113031b4f15a3c1104b5f135e8346997 # v0.0.6
+        with:
+          head: {% raw %}${{ github.sha }}{% endraw %}
+          base: {% raw %}${{ github.event.pull_request.base.sha || github.event.merge_group.base_sha || 'origin/master' }}{% endraw %}
+          n_parents: 0
+          exclude_tags: "gpu"
+
+      - name: Separate modules and subworkflows
+        id: components
+        run: |
+          echo modules=$(echo {% raw %}'${{ steps.list.outputs.components }}'{% endraw %} | jq -c '. | map(select(contains("modules"))) | map(gsub("modules/{{ short_org_name }}/"; ""))') >> $GITHUB_OUTPUT
+          echo subworkflows=$(echo {% raw %}'${{ steps.list.outputs.components }}'{% endraw %} | jq '. | map(select(contains("subworkflows"))) | map(gsub("subworkflows/{{ short_org_name }}/"; ""))') >> $GITHUB_OUTPUT
+
+      - name: get number of shards
+        id: set-shards
+        if: {% raw %}${{ steps.list.outputs.components != '[]' }}{% endraw %}
+        uses: ./.github/actions/get-shards
+        env:
+          NFT_VER: {% raw %}${{ env.NFT_VER }}{% endraw %}
+        with:
+          max_shards: 15
```

```suggestion
          max_shards: 15
```
not sure how high this should be for a standard set of github runners (which should be 30 I think, so would  quite easily fill up).

---

If I read correctly we have 60 runners available, but we're on the Teams plan. 30 is perhaps the default ?

Happy to reduce it to say 10 or 5 here.